src/component/hooks/usePanelPreferences.ts

```diff
@@ -0,0 +1,11 @@
+import { useMemo } from 'react';
+
+import { usePreferences } from '../context/PreferencesContext';
+
+export function usePanelPreferences(key: string) {
+  const preferences = usePreferences();
+
+  return useMemo(() => {
+    return preferences.formatting.panels[key] || null;
+  }, [key, preferences.formatting.panels]);
```

Comment:
nothing takes time or risks creating a new object here, the useMemo is counterproductive (comparing the dependencies is not costless)